### PR TITLE
Add support for displaying author information from Goodreads

### LINKED ISSUE #211: [Enhancement] Author bio page by Wikipedia
Hi, a suggestion:
make an author bio page so that when we click on the author name we get their books and at the same page a biography and a photo of the author.

### PR DESCRIPTION
**Pull request by jkrehm**
Retrieves Goodreads author information and displays their photo and "about" text.

Requires an API key and the "goodread" module (added to optional-requirements.txt).

Currently uses https://github.com/sefakilic/goodreads which is self-proclaimed to be unmaintained (but works without issue). If ever it becomes an issue, https://github.com/tatianass/goodreads2 has been actively worked on recently.

Resolves #211

## Configuration
screen shot 2017-07-08 at 4 16 04 pm

## Author Page
screen shot 2017-07-08 at 4 12 23 pm

### CODE REVIEW

**Comment by janeczku** [comment]
@jkrehm The Goodread integration looks great. Could you please remove and open a separate PR for the Google Drive patches you included in this PR? Thanks!

**Comment by jkrehm** [comment]
Certainly. I didn't really want to lump them together, but one part of the Google Drive patch depends on work done for this ticket. I've created a [branch](https://github.com/jkrehm/calibre-web/tree/gdrive-changes) in my fork for the fixes and will rebase and send a PR if/when this one is accepted.

Thanks for looking it over!

File: cps/templates/author.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{% extends "layout.html" %}
+{% block body %}
+<h2>{{title}}</h2>
+
+{% if author is not none %}
+<section class="author-bio">
+  {%if author['image_url'] is not none %}
```
Comment (janeczku) [inline cps/templates/author.html]: `author` object is not a dict. Causes:

```
[2017-07-14 02:06:11,584] ERROR in app: Exception on /author/86 [GET]
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Library/Python/2.7/site-packages/flask_login/utils.py", line 228, in decorated_view
    return func(*args, **kwargs)
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/web.py", line 1114, in author
    title=name, author=author_info)
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/web.py", line 587, in render_title_template
    return render_template(instance=config.config_calibre_web_title, *args, **kwargs)
  File "/Library/Python/2.7/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/Library/Python/2.7/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/Library/Python/2.7/site-packages/jinja2/environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Library/Python/2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/templates/author.html", line 1, in top-level template code
    {% extends "layout.html" %}
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/templates/layout.html", line 168, in top-level template code
    {% block body %}{% endblock %}
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/templates/author.html", line 7, in block "body"
    {%if author['image_url'] is not none %}
  File "/Library/Python/2.7/site-packages/jinja2/environment.py", line 378, in getitem
    return obj[argument]
AttributeError: GoodreadsAuthor instance has no attribute '__getitem__'
```

File: cps/templates/author.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  <img src="{{author.image_url}}" alt="{{author.name}}" class="author-photo pull-left">
```
Comment (janeczku) [inline cps/templates/author.html]: When `author` attributes contains unicode characters (e.g. name="Honoré de Balzac") an exception occurs. 
I could fix it by replacing any occurrence of `{{author.name}}` with `{{author.name.decode('utf-8')}}`.

```
InsecurePlatformWarning
[2017-07-14 02:08:58,027] ERROR in app: Exception on /author/120 [GET]
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Library/Python/2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Library/Python/2.7/site-packages/flask_login/utils.py", line 228, in decorated_view
    return func(*args, **kwargs)
  File "/Users/Janek/Downloads/calibre/calibre-web/cps/web.py", line 1109, in author
    print author_info
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 5: ordinal not in range(128)
ERROR:tornado.access:500 GET /author/120 (::1) 2652.11ms
```

File: cps/templates/author.html (hunk 1 above)
Comment (jkrehm) [inline cps/templates/author.html]: Ah, missed changing that reference to `author.image_url`.

File: cps/templates/author.html (hunk 2 above)
Comment (jkrehm) [inline cps/templates/author.html]: Weird that Jinja2 doesn't take care of that already. Don't book titles also have a problem? If not, do you know why not?

Comment (jkrehm) [inline cps/templates/author.html]: Ah, I think it's an issue with the Goodreads library. Shame. https://github.com/sefakilic/goodreads/issues/7

Comment (jkrehm) [inline cps/templates/author.html]: I'm not able to recreate the issue. I downloaded a book by Honoré de Balzac, added it to my library, and without any changes the page loads without an error.

**Comment by jkrehm** [comment]
Not sure how I mark it ready to go, but I made the requested changes. I wasn't able to recreate the issue with unicode characters in the author's name, so I'm only assuming the change will solve the problem.

**Comment by jkrehm** [comment]
Thanks!

**Pull request merged**